### ISSUE TITLE
Api documentation

### ISSUE THREAD

**Issue by kfh** [adamko-dev/kotka-streams#27]
Just discovered this nice little library and while playing around i thought it would be useful to have docs of the full api available. What about using [Dokka](https://github.com/Kotlin/dokka) for a nice viewing experience ?

**Comment by aSemy** [comment]
Hi, thanks for the interest!

Publishing docs is something I want to do.

I've been working on setting up a GitHub Pages site, with demonstrations of functionality, that are automatically tested. The result would be similar to https://adamko-dev.github.io/kotlinx-serialization-typescript-generator/. Would that suit your needs?

I'm not a big fan of Dokka. It's slow, and the Gradle plugin [doesn't follow the Gradle API correctly](https://github.com/Kotlin/dokka/issues/2700) which causes build problems. I think that publishing the sources JAR is good enough for usage within IDEs.

**Comment by kfh** [comment]
Thanks for the feedback. I just wanna get a quick view of the API without opening an editor. Whether i'm on the phone or in front of my computer. I do like how Dokka presents the code but if you have other solutions in mind that's not a big issue for me.

**Comment by aSemy** [comment]
Hey @kfh, I've just finished with the API docs

https://adamko-dev.github.io/kotka-streams/

What do you think?

**Comment by kfh** [comment]
> Hey @kfh, I've just finished with the API docs
> 
> 
> 
> https://adamko-dev.github.io/kotka-streams/
> 
> 
> 
> What do you think?

Yeah, this looks good 👌 Exactly what i felt was missing. Great work
